packages/toolkit/src/view/knowledge/components/StatusTag.tsx

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import { Tag } from "@instill-ai/design-system";
+
+type StatusTagProps = {
+    status: string;
+};
+
+type FileStatus =
+    | "NOTSTARTED"
+    | "WAITING"
+    | "CONVERTING"
+    | "CHUNKING"
+    | "EMBEDDING"
+    | "COMPLETED"
+    | "FAILED";
+
+type TagVariant =
+    | "lightNeutral"
+    | "lightYellow"
+    | "default"
+    | "lightGreen"
+    | "lightRed";
+
+const getStatusTag = (
+    status: FileStatus
+): { variant: TagVariant; dotColor: string } => {
+    const statusMap: Record<
+        FileStatus,
+        { variant: TagVariant; dotColor: string }
+    > = {
+        NOTSTARTED: {
+            variant: "lightNeutral",
+            dotColor: "bg-semantic-fg-secondary",
+        },
+        WAITING: {
+            variant: "lightYellow",
+            dotColor: "bg-semantic-warning-default",
+        },
+        CONVERTING: { variant: "default", dotColor: "bg-semantic-accent-default" },
+        CHUNKING: { variant: "default", dotColor: "bg-semantic-accent-default" },
+        EMBEDDING: { variant: "default", dotColor: "bg-semantic-accent-default" },
+        COMPLETED: {
+            variant: "lightGreen",
+            dotColor: "bg-semantic-success-default",
+        },
+        FAILED: { variant: "lightRed", dotColor: "bg-semantic-error-default" },
```

Comment:
We don't use TailwindCSS native color in this codebase, please consult Dani for correct color